mozilla_version/gecko.py

```diff
@@ -146,18 +149,20 @@ class GeckoVersion(object):
             |(?P<is_aurora_or_devedition>a2)
             |b(?P<beta_number>\d+)
             |(?P<is_esr>esr)
+            |(?P<is_release_candidate>rc)(?P<rc_number>\d*)
         )?
         -?(build(?P<build_number>\d+))?$""", re.VERBOSE)
 
     _ALL_VERSION_NUMBERS_TYPES = (
-        'major_number', 'minor_number', 'patch_number', 'beta_number',
+        'major_number', 'minor_number', 'patch_number', 'beta_number', 'rc_number'
     )
 
     major_number = attr.ib(type=int, converter=_positive_int)
     minor_number = attr.ib(type=int, converter=_positive_int)
     patch_number = attr.ib(type=int, converter=_positive_int_or_none, default=None)
     build_number = attr.ib(type=int, converter=_strictly_positive_int_or_none, default=None)
     beta_number = attr.ib(type=int, converter=_strictly_positive_int_or_none, default=None)
+    rc_number = attr.ib(type=int, converter=_strictly_positive_int_or_none, default=None)
```

Comment:
That's an interesting approach! So far, I've always seen `62.0rc2` == `62.0build2`. Do you think merging the `rc_number` and  the `build_number` always makes sense? If not, what edge case do you think of?

Reply:
If we put in 'rc2' or 'build2' we should definitely get the same string back out again. I should definitely fix up the PR for comparison operators so that they're equivalent. Perhaps we could store it internally as build number, and use a flag to say 'display this as `rc`'?

Reply:
So with this option, it would be changing `|(?P<is_release_candidate>rc)(?P<rc_number>\d*)` to `|(?P<is_release_candidate>rc)(?P<build_number>\d*)` and using the stored `is_release_candidate` to control the representation in `__str__`. I think that does make sense, if we're happy they're always going to be equivalent

Reply:
Yeah, that's a great approach! I'm all for it :+1:

Reply:
Do we currently generate version numbers with`rcN` anywhere programmatically anywhere? I wonder if it would make more sense to have a method that generates the `rcN` format in those places that need it, rather than controlling that based on what format we happened to parse.

Reply:
It would mean having a method that's invalid for 5/6 of the version types, which is concerning. I'm unsure if we programmatically generate `rcN`, though.

Reply:
>  It would mean having a method that's invalid for 5/6 of the version types, which is concerning. I'm unsure if we programmatically generate `rcN`, though.

It wouldn't be invalid, just not differ from `__str__` on most of them.